trace4cats/src/main/scala/prometheus4cats/trace4cats/Trace4CatsExemplarSampler.scala

```diff
@@ -0,0 +1,161 @@
+      spanIdLabelName: Exemplar.LabelName
+  ) extends Trace4CatsExemplarSampler[F, A](traceIdLabelName, spanIdLabelName)
+      with ExemplarSampler[F, A] {
+
+    protected def sampleImpl(
+        value: A,
+        buckets: NonEmptySeq[Double],
+        previous: Option[Exemplar.Data],
+        next: Option[Exemplar.Data]
+    ): Option[Exemplar.Data]
+
+    protected def sampleImpl(
+        previous: Option[Exemplar.Data],
+        next: Option[Exemplar.Data]
+    ): Option[Exemplar.Data]
+
+    protected def sampleImpl(
+        value: A,
+        previous: Option[Exemplar.Data],
+        next: Option[Exemplar.Data]
+    ): Option[Exemplar.Data]
+
+    override def sample(
+        value: A,
+        buckets: NonEmptySeq[Double],
+        previous: Option[Exemplar.Data]
+    ): F[Option[Exemplar.Data]] =
+      spanExemplarData.map(sampleImpl(value, buckets, previous, _))
+
+    override def sample(
+        previous: Option[Exemplar.Data]
+    ): F[Option[Exemplar.Data]] = spanExemplarData.map(sampleImpl(previous, _))
+
+    override def sample(
+        value: A,
+        previous: Option[Exemplar.Data]
+    ): F[Option[Exemplar.Data]] =
+      spanExemplarData.map(sampleImpl(value, previous, _))
+  }
+
+  abstract class Simple[F[_]: FlatMap: Clock: Trace.WithContext, A](
+      traceIdLabelName: Exemplar.LabelName,
+      spanIdLabelName: Exemplar.LabelName
+  ) extends FromTraceContext[F, A](traceIdLabelName, spanIdLabelName) {
+    override protected def sampleImpl(
+        value: A,
+        buckets: NonEmptySeq[Double],
+        previous: Option[Exemplar.Data],
+        next: Option[Exemplar.Data]
+    ): Option[Exemplar.Data] = sampleImpl(previous, next)
+
+    override protected def sampleImpl(
+        value: A,
+        previous: Option[Exemplar.Data],
+        next: Option[Exemplar.Data]
+    ): Option[Exemplar.Data] = sampleImpl(previous, next)
+  }
+
+  class Default[
+      F[_]: FlatMap: Clock: Trace.WithContext,
+      A
+  ](
+      minRetentionInterval: FiniteDuration,
+      traceIdLabelName: Exemplar.LabelName,
+      spanIdLabelName: Exemplar.LabelName
+  ) extends Simple[F, A](traceIdLabelName, spanIdLabelName) {
+    private val minRetentionIntervalMs = minRetentionInterval.toMillis
+
+    override protected def sampleImpl(
+        previous: Option[Exemplar.Data],
+        next: Option[Exemplar.Data]
+    ): Option[Exemplar.Data] = previous match {
+      case Some(prev) =>
+        next
+          .filter(
+            _.timestamp.toEpochMilli - prev.timestamp.toEpochMilli > minRetentionIntervalMs
+          )
+          .orElse(previous)
+      case None => next
+    }
+  }
+
+  // Choosing a prime number for the retention interval makes behavior more predictable,
+  // because it is unlikely that retention happens at the exact same time as a Prometheus scrape.
+  val DefaultMinRetentionInterval: FiniteDuration = 7109.millis
+
+  def apply[F[_]: FlatMap: Clock: Trace.WithContext, A](
```

Comment:
**suggestion:** maybe we could rename this to `default`? 

<sub>This comment follows the [**conventionalcomments.org**](https://conventionalcomments.org) standard</sub>

Reply:
Sounds good!